tabular/src/autogluon/tabular/models/rf/compilers/onnx.py

```diff
@@ -59,22 +60,53 @@ def can_compile():
             return False
 
     @staticmethod
-    def compile(obj, path: str):
-        if isinstance(obj.model, RFOnnxPredictor):
-            return obj.model
-        # Convert into ONNX format
+    def compile(model, path: str, input_types=None):
+        """
+        Compile the trained model for faster inference.
+
+        Parameters
+        ----------
+        model
+            The native model that is expected to be compiled.
+        path : str
+            The path for saving the compiled model.
+        input_types : list, default=None
+            A list of tuples containing shape and element type info, e.g. [((1, 14), np.float32),].
+            The list would be used as the input data for the model.
+            The compiler would optimize the model to perform best with the given input type.
+        """
+        if input_types is None or not isinstance(input_types[0], tuple):
+            raise RuntimeError("input_types argument should contain at least one tuple"
+                               ", e.g. [((1, 14), np.float32)]")
+        if isinstance(model, RFOnnxPredictor):
+            return model
+
         from skl2onnx import convert_sklearn
         from skl2onnx.common.data_types import FloatTensorType
-        initial_type = [('float_input', FloatTensorType([None, obj._num_features_post_process]))]
-        onx = convert_sklearn(obj.model, initial_types=initial_type)
+        from sklearn.ensemble import RandomForestClassifier, ExtraTreesClassifier
+        input_shape = list(input_types[0][0])
+        initial_type = [('float_input', FloatTensorType(input_shape))]
+
+        # Without ZipMap
+        # See http://onnx.ai/sklearn-onnx/auto_examples/plot_convert_zipmap.html#without-zipmap
+        options = {}
+        if isinstance(model, (RandomForestClassifier, ExtraTreesClassifier)):
+            options = {id(model): {'zipmap': False}}
+
+        # Convert the model to onnx
+        onnx_model = convert_sklearn(model, initial_types=initial_type, options=options)
+        predictor =  RFOnnxPredictor(model=onnx_model)
+        RFOnnxCompiler.save(onnx_model, path)
+
+    @staticmethod
+    def save(model, path: str):
```

Comment:
add type hints for output, ditto elsewhere

Reply:
Added documents and output type hints.